CONTRIBUTING.md

```diff
@@ -0,0 +1,146 @@
+# Contributing to Tlog
+The Tlog project welcomes new contributors. This document will guide you through the process.
+
+## Pull Requests
+We actively welcome your pull requests. There are a few guidelines that we need
+contributors to follow so that we can keep the codebase consistent and clean.
+
+* Fork the main repository and clone it to your local machine.
+* Create a topic branch from master. Please avoid working directly on the ```master``` branch.
+* Make commits of logical units, and make sure to follow the CODINGSTYLE.
+* Submit the Pull Request(PR) on Github.
+
+## Issues
+We use GitHub [issues tracker](https://github.com/Scribery/tlog/issues) to track bugs.
+Please ensure your description is clear and has sufficient instructions; so we are able to reproduce the issue.
+
+## Coding Style guidelines
+
+* Use four spaces for indentation, please don't use tabs except where required, such as in Makefiles.
+
+* Line should be at most 78 chars.
+
+* Please don't leave trailing whitespace at the ends of lines.
+
+* Braces open on the same line as the `for`/`while`/`if`/`else`/`switch`/etc. Closing
+  braces are put on a line of their own, except in the `else` of an `if` statement
+  or in a while of a `do`-`while` statement. Always use braces for `if` and `while`.
+
+```c
+if (a == b) {
+    ...
+}
+
+if (a == b) {
+    ...
+} else if (a > b) {
+    ...
+}
+
+if (a == b) {
+    do_something();
+} else {
+    do_something_else();
+}
+
+do {
+    do_something();
+} while (cond);
+
+while (cond) {
+    do_something();   
+}
+
+switch (expression) {
+case 'constant-expression':
+    ...
+case 'constant-expression':
```

Comment:
I think `constant-expression` shouldn't be quoted here. It doesn't look valid C syntax.